src/main/java/com/shank/offcoder/cf/QuestionManager.java

```diff
@@ -74,21 +75,17 @@ public static void deleteQuestion(Controller controller, final List<ProblemParse
         new Thread(() -> {
             JSONArray arr = AppData.get().getData(AppData.DOWNLOADED_QUES, new JSONArray());
             ArrayList<Integer> questionIndex = new ArrayList<>();
+            HashSet<String> pCode = new HashSet<>();
+            for (ProblemParser.Problem p : list) {
+                pCode.add(p.code);
+            }
             double counter = 0;
-            int idx = -1;
-            for (Object j : arr) {
-                idx++;
-                boolean flag = false;
-                for (ProblemParser.Problem p : list) {
-                    if (j.toString().contains(p.code)) {
-                        ++counter;
-                        controller.downloadProgress.setProgress(counter / list.size());
-                        flag = true;
-                        break;
-                    }
-                }
-                if (flag) {
-                    questionIndex.add(idx - ((int) counter - 1));
+            for (int i = 0; i < arr.length(); i++) {
+                JSONObject item = arr.getJSONObject(i);
+                if (pCode.contains(item.get(AppData.P_CODE_KEY).toString())) {
+                    ++counter;
+                    controller.downloadProgress.setProgress(counter / list.size());
+                    questionIndex.add(i);
```

Comment:
@Aqua-16 how about `arr.remove(i)` here itself ?

Reply:
Won't that cause an error? I haven't tried that yet, but doing something like that usually gives me an error.